tests/schedule.py

```diff
@@ -24,13 +24,13 @@ def testContains(self):
         self.assertFalse('NOT' in self.sch)
 
     def testStartEnd(self):
-        self.assertEqual(dt.date(2015, 1, 1),   self.sch.start)
-        self.assertEqual(dt.date(2029, 12, 27), self.sch.end)
+        self.assertEqual(dt.date(2015, 1, 1),   self.sch.start.date())
+        self.assertEqual(dt.date(2029, 12, 27), self.sch.end.date())
 
     def testTimesteps(self):
         timesteps = self.sch.timesteps
         self.assertEqual(176, len(timesteps))
-        self.assertEqual(dt.date(2015, 12, 31), timesteps[7])
+        self.assertEqual(dt.date(2015, 12, 31), timesteps[7].date())
```

Comment:
please see if you can do this without changing the API

Reply:
We can probably do this without changing the API, but the API was in my opionon slightly broken on this particular point - the timepoints given by the `Schedule` object were in the form of `python datetime.date` instances, but the underlying datamodel has second resolution so python `datetime.datetime` seems more appropriate.

Reply:
Sounds good, please do that.

Reply:
> please do that.

Please do what?

Reply:
See if you can complete the refactoring without changing the tests?  Or isn't `timesteps[7]` supposed to be a date?

Reply:
If not, then we should change the left-hand side to contain the hour/minute/seconds part.

Reply:
> Or isn't timesteps[7] supposed to be a date?

No timestep is supposed to be full datetime, that was I why i considered the previous API to be broken. 

> then we should change the left-hand side to contain the hour/minute/seconds part.

I agree that is a better fix - will do that.

Reply:
> I agree that is a better fix - will do that.

Done. I let the the code changing the test be in a separate commit - so that it would stand out clearly for review; but I guess it should really be melded together with the main commit?